design/sig.txt

```diff
@@ -24,56 +26,77 @@ ways the MPS detects corruption or the passing of illegal data is using
 *signatures*.  This simple technique has proved invaluable at catching
 defects early.
 
-This is based on [RB_1995-08-25]_.
+This document is based on [RB_1995-08-25]_.
 
 
 Overview
 --------
-Signatures are `magic numbers`_ which are written into structures when
-they are created and invalidated (by overwriting with ``SigInvalid``)
-when they are destroyed. They provide a limited form of run-time type
-checking and dynamic scope checking. They are a simplified form of
-"Structure Marking", a technique used in the Multics filesystem
-[THVV_1995]_.
+
+_`.overview`: Signatures are `magic numbers`_ which are written into
+structures when they are created and invalidated (by overwriting with
+``SigInvalid``) when they are destroyed. They provide a limited form
+of run-time type checking and dynamic scope checking. They are a
+simplified form of "Structure Marking", a technique used in the
+Multics filesystem [THVV_1995]_.
 
 .. _`magic numbers`: https://en.wikipedia.org/wiki/Magic_number_(programming)
 
 
 Definitions
 -----------
-Nearly every structure should start with a field of type ``Sig`` with the name
-``sig``.  For example::
+
+_`.field`: Nearly every structure should start with a field of type
+``Sig`` with the name ``sig``.  For example::
 
     typedef struct mps_message_s {
-      Sig sig;                      /* <design/sig/> */
+      Sig sig;                      /* design.mps.sig */
       Arena arena;                  /* owning arena */
       MessageClass class;           /* Message Class Structure */
       Clock postedClock;            /* mps_clock() at post time, or 0 */
       RingStruct queueRing;         /* Message queue ring */
     } MessageStruct;
 
-There must also be a definition for the valid value for that signature::
+_`.value`: There must also be a definition for the valid value for
+that signature::
```

Comment:
m: we do not specify whether sigs should be unique to classes. The use case of run time checking implies they might be, however 2 similarly named classes could end up with identical sigs with different meanings. We should decide if we need a rule about this, either to impose uniqueness or we might equally decide we do not want to impose such a limitation on a developer.